main.go

```diff
@@ -387,20 +394,36 @@ func main() {
 	instanceName, _ := os.Hostname() // on an error, instanceName will be empty, which is ok
 
 	var nsSelReconciler common.NamespaceSelectorReconciler
+	var nsSelUpdatesSource *source.Channel
+	var objectTemplatesChannel *source.Channel
+	var dynamicWatcher depclient.DynamicWatcher
 	var dryRunSupported bool
+	var serverVersion *k8sversion.Info
+
+	managerCtx, managerCancel := context.WithCancel(terminatingCtx)
 
+	// Don't initialize any clients that use the target clusters when being uninstalled. This is to guard against
+	// the hosted cluster being gone during uninstalls when running in hosted mode.
 	if !beingUninstalled {
-		nsSelReconciler = common.NamespaceSelectorReconciler{
-			Client: nsSelMgr.GetClient(),
+		// Set the buffer to 20 to not take up too much memory. If the buffer get's filled, that's okay because the
+		// recipient is really just parsing the event and adding it to a queue, so it should get capacity quickly.
+		nsSelUpdatesChan := make(chan event.GenericEvent, 20)
+		nsSelUpdatesSource = &source.Channel{
+			Source:         nsSelUpdatesChan,
+			DestBufferSize: 20,
 		}
+
+		nsSelReconciler = common.NewNamespaceSelectorReconciler(nsSelMgr.GetClient(), nsSelUpdatesChan)
 		if err = nsSelReconciler.SetupWithManager(nsSelMgr); err != nil {
 			log.Error(err, "Unable to create controller", "controller", "NamespaceSelector")
 			os.Exit(1)
 		}
 
 		discoveryClient := discovery.NewDiscoveryClientForConfigOrDie(targetK8sConfig)
 
-		serverVersion, err := discoveryClient.ServerVersion()
+		var err error
```

Comment:
I like this since it doesn't overwrite the `err` value from the above scope.

Reply:
So it shadows the old `err`, instead of overwriting it? Maybe this is another case where it just needs a different name.

But this is pretty minor - we can keep it as-is if you really prefer.

Reply:
Actually, if I'm looking at it right, I think this is correct. The old code was also scoping `serverVersion` such that it would not be populated for the `reconciler` variable farther down...